libnmstate/nm/applier.py

```diff
@@ -48,16 +48,17 @@ def prepare_new_ifaces_configuration(ifaces_desired_state):
 
 def edit_existing_ifaces(con_profiles):
     for connection_profile in con_profiles:
-        devname = connection_profile.get_interface_name()
+        devname = connection_profile.devname
```

Comment:
> It is hard to understand why this code makes sense - if it does. AFAIU

We are talking about a syntactic change of objectizing the connection entity.
Nothing should have changed logically, your comment on this are just, but should be addressed through either a separate PR (refactoring) or through a ticket.

> this indicates that the connection profile object should return a device object IMHO

That creates a dependency which is not needed between the two (connection and device).
Anyway, that is just an addition which can be added later when we will have a device object wrapper.

> I do not really get the idea behind this or why this complicated flow is needed.

This is just an ugly way to check if the the profile in question is of a new device/profile or of an existing one.
It is a subsequence of using two stages (the first being `prepare_edited_ifaces_configuration`).

Hopefully this can be fixed and refactor in the near future, probably as apart of merging the two mainloop runs as we already discussed in the past.

Reply:
> This is just an ugly way to check if the the profile in question is of a new device/profile or of an existing one.
> It is a subsequence of using two stages (the first being `prepare_edited_ifaces_configuration`).

I see. This could also solved inside the object so that it knows whether or not it is new or changed and calling ConnectionProfile.commit() will just do the right thing without this complicated check.